test/nodejs/test_io.rb

```diff
@@ -15,4 +15,10 @@ def test_binread_noexistent_should_raise_io_error
       IO.binread('tmp/nonexistent')
     end
   end
+
+  def test_binread_encoding
+    File.write('tmp/foo', 'Le français c\'est compliqué :)\n')
+    assert_equal("Le fran\xC3\xA7ais c'est compliqu\xC3\xA9 :)\\n", IO.binread('tmp/foo'))
```

Comment:
I'm pretty sure that this expectation fails because Ruby strings become JS strings (usually) without any changes:
```
$ node
> "Le fran\xC3\xA7ais c'est compliqu\xC3\xA9 :)\\n"
'Le franÃ§ais c\'est compliquÃ© :)\\n'
> "Le français c'est compliqué :)\\n"
'Le français c\'est compliqué :)\\n'
```

Reply:
Maybe something like:
``` ruby
content = 'Le français c\'est compliqué :)\n'
File.write('tmp/foo', content)
written = IO.binread('tmp/foo')
assert_equal(written, content)
assert_equal(written.encoding, Encoding::ASCII_8BIT)
```

(requires adding `.force_encoding('ascii-8bit')` to the `IO.binread` method).

Reply:
My bad, `parser` handled it for us:
```
$ cat sample.rb
assert_equal("Le fran\xC3\xA7ais c'est compliqu\xC3\xA9 :)\\n", IO.binread('tmp/foo'))
$ ruby-parse sample.rb
(send nil :assert_equal
  (str "Le français c'est compliqué :)\\n")
  (send
    (const nil :IO) :binread
    (str "tmp/foo")))
```
These two expectations are just equal.

Reply:
Yes I added both `"Le fran\xC3\xA7ais c'est compliqu\xC3\xA9 :)\\n"` and `"Le français c'est compliqué :)\\n"`